### ISSUE TITLE
Add ability for users to close polls; configurable opening/closing time for polls

### ISSUE THREAD

**Issue by mudlock** [electology/approval_frame#15]

**Comment by Brianjr0428** [comment]
@ashimaathri 
I am looking into this issue and I have several questions:

- If my understanding is correct, we need to allow the creator remove the poll he created, and this should be done in the detail page. Is that correct?
- what does 'configurable opening/closing time for polls' means here?
- Anything else I need to pay attention?

Thanks!

**Comment by ashimaathri** [comment]
Hi @Brianjr0428. This issue depends on #14 after which the user will have a home page which lists their polls. The user can then manage the polls from the home page (remove, suspend, edit options etc.)

Configurable opening/closing time means that the poll is open for voting only until a specific time (which is configured by the poll owner). After that only the results of the poll can be seen but a user can not vote in that poll.

**Comment by Brianjr0428** [comment]
Hi @ashimaathri,
How about I try to solve issue #14 first? I will try to create a page for an individual user to review their polls and manage them.
If you think it's okay for me to try it, could you tell me specific requirement for it?
Please let me know how you think. Thanks.

**Comment by ashimaathri** [comment]
@Brianjr0428 That would be great! I can't think of any specific requirement but if you get stuck with something, let me know.

**Comment by Brianjr0428** [comment]
@ashimaathri 
I think I am ready to take this one and I have some questions to clarify:
- Can you give me more detailed requirements for the functionality of suspend and edit?
- For the remove option, do you just want to simply remove the poll itself when clicking it or show a warning message such as review page or a pop-up window?
Thanks!

**Comment by ashimaathri** [comment]
Great!

* Suspend should basically make the poll 'unvote-able' but not remove it from the database. Meaning that routes to it should return 404 and it should not show up on the index page but it should still show up on the users's 'Manage Polls' page (in case the user wants to reactivate it)
* Edit should open a form where the user can change attributes of the poll like closing date, question, choices etc.
* I think some kind of popup/warning is best to prevent accidental deletion of polls.

**Comment by Brianjr0428** [comment]
This one looks big to me since it involves three changes.
In that case, I would like to divide it into three sub-issues and do PR respectively.
How to you think?

**Comment by ashimaathri** [comment]
@Brianjr0428 Sounds good! Could you create the three sub issues too? Thanks!

**Comment by Brianjr0428** [comment]
@ashimaathri 
I've created three new issue #65 #66 #67.
Just my personal thought, do we still need this issue open?
I will be out of town this week and would be a little busy in for a couple of weeks, but I will find some time to do them.
Thanks.

**Comment by ashimaathri** [comment]
Thanks @Brianjr0428. We can close issue #15 when we close all the three that you've created (less confusing that way).
Take your time and hope you have a great trip!

**Comment by Brianjr0428** [comment]
@ashimaathri 
Excuse me, I tried to work it today and pull the code from upstream master(when the last PR happened), but it seems the CSS has gone as below:
![65](https://cloud.githubusercontent.com/assets/4754126/8560499/3a6739f6-24cd-11e5-8e5e-09db8002f1db.png)
Is it what it's supposed to be? Or did I do anything wrong?
I am thinking working on the last changes of issues14 but I am afraid it will cause inconsistency and I will have to fix it i the future.
Thanks

**Comment by ashimaathri** [comment]
Yes, I noticed this too. After some digging, I found it to be because of issue #62. For now, just set DEBUG=True in settings.py or local_settings.py and everything should work fine.

**Comment by Daniel-Avila** [comment]
This was last updated 26 days ago. I would like to take a swing at this.

**Comment by ashimaathri** [comment]
@Daniel-Avila we split this issue into 3 parts and @Brianjr0428 is looking into the first part #65. It would be great if you could look into one of the other two (or maybe both :smile:  ) #66, #67

**Comment by Daniel-Avila** [comment]
Sure thing!

**Comment by rupalkhilari** [comment]
This was added as a part of #81. Closing this.
